File: 1-two-sum/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        Dictionary<int, int> numsDict = new(); 
        
        for(int a = 0; a < nums.Length; a++) {
            if (numsDict.TryGetValue(target - nums[a], out int b)){
                return new int[]{b,a};
            }
            numsDict[nums[a]] = a;
        }
        return null;
    }
}

File: 1-two-sum/1-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    
    //Storing the value and the index 
    const numsMap = new Map<number, number>();
    
    let returnNums: number[] = [];
    
    for (let i = 0; i < nums.length; i++) {
        const targetIndex = numsMap.get(target - nums[i]);
        
        if (targetIndex !== undefined) {
            returnNums = [targetIndex, i];
            break;
        }
        
        numsMap.set(nums[i], i);
    }
    
    return returnNums;

};